runtime/connections.go

```diff
@@ -290,31 +287,23 @@ func (r *Runtime) ConnectorConfig(ctx context.Context, instanceID, name string)
 	return res, nil
 }
 
-// ResolveConnectorProperties resolves templating in the provided connector's properties.
+// resolveConnectorProperties resolves templating in the provided connector's properties.
 // It always returns a clone of the properties, even if no templating is found, so the output is safe for further mutations.
-func ResolveConnectorProperties(environment string, vars map[string]string, c *runtimev1.Connector) (map[string]string, error) {
-	res := maps.Clone(c.Config)
+func resolveConnectorProperties(environment string, vars map[string]string, c *runtimev1.Connector) (map[string]any, error) {
+	res := c.Properties.AsMap()
 	if res == nil {
-		res = make(map[string]string)
+		res = make(map[string]any)
 	}
-
-	// Resolve templating in properties that use it
-	for _, k := range c.TemplatedProperties {
```

Comment:
The change here drops this optimization. Is there an easy way we can keep it, at least for root-level props? This code is very hot path (almost every API call), so nice not to run the template engine redundantly